pkg/virt-controller/network/pod_annotations_test.go

```diff
@@ -0,0 +1,151 @@
+package network_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/pkg/network/downwardapi"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/deviceinfo"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/network"
+)
+
+var _ = Describe("pod annotations", func() {
+
+	Context("Generate pod network annotations", func() {
+		const (
+			deviceInfoPlugin    = "deviceinfo"
+			nonDeviceInfoPlugin = "non_deviceinfo"
+		)
+
+		bindingPlugins := map[string]virtv1.InterfaceBindingPlugin{
+			deviceInfoPlugin:    {DownwardAPI: virtv1.DeviceInfo},
+			nonDeviceInfoPlugin: {},
+		}
+
+		networkStatus := `
+[
+{
+ "name": "default/no-device-info",
+ "interface": "pod6446d58d6df",
+ "mac": "8a:37:d9:e7:0f:18",
+ "dns": {}
+},
+{
+ "name": "default/with-device-info",
+ "interface": "pod2c26b46b68f",
+ "dns": {},
+ "device-info": {
+   "type": "pci",
+   "version": "1.0.0",
+   "pci": {
+     "pci-address": "0000:65:00.2"
+   }
+ }
+},
+{
+ "name": "default/sriov",
+ "interface": "pod778c553efa0",
+ "dns": {},
+ "device-info": {
+   "type": "pci",
+   "version": "1.0.0",
+   "pci": {
+     "pci-address": "0000:65:00.3"
+   }
+ }
+}
+]`
```

Comment:
nit: indentation would be nice, also it seems it can defined as const
```
networkStatus := `[
      {
        "name": "default/no-device-info",
        "interface": "pod6446d58d6df",
        "mac": "8a:37:d9:e7:0f:18",
        "dns": {}
      },
      {
        "name": "default/with-device-info",
        "interface": "pod2c26b46b68f",
        "dns": {},
        "device-info": {
          "type": "pci",
          "version": "1.0.0",
          "pci": {
            "pci-address": "0000:65:00.2"
          }
        }
      },
      {
        "name": "default/sriov",
        "interface": "pod778c553efa0",
        "dns": {},
        "device-info": {
          "type": "pci",
          "version": "1.0.0",
          "pci": {
            "pci-address": "0000:65:00.3"
          }
        }
      }
  ]`
```